### PR TITLE
Diet Recommendation for Pregnancy

### PR DESCRIPTION
**Pull request by Nabanita29**
##  Info about the related issue
- [x] GSSOC Participant
- [x] Contributor

Closes: #766 

My solution involves collecting a comprehensive dataset with pregnant women's characteristics, pregnancy stage, dietary restrictions, nutritional needs, and food items with nutritional information. The data has been preprocessed to handle missing values, encode categorical variables, and scale numerical features. Multiple multi-label classification models, including Random Forest and neural network model, has been evaluated and trained on the dataset. Hyperparameter tuning has also been performed to enhance model performance.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots
image
image
image

### CODE REVIEW

**Comment by Nabanita29** [comment]
Ma'am I will update the readme and dataset if you can review and let me know if this project will be accepted!
